test/interactiveutil.jl

```diff
@@ -0,0 +1,110 @@
+### Testing download(url,[localfile],[silent=false])
+
+"Runs an expression, and returns stdout, stderr, and the result."
+function capture_output(expression)
+	originalSTDOUT = STDOUT
+	originalSTDERR = STDERR
+	(errRead, errWrite) = redirect_stderr()
+	(outRead, outWrite) = redirect_stdout()
+
+	result = eval(expression)	
+	# Ensure both streams are written to or readavailable waits forever.
+	print("\n")
+	write(STDERR, "\n")
+
+	close(errWrite)
+	close(outWrite)
+	errors = readavailable(errRead)
+	output = readavailable(outRead)
+	close(outRead)
+	close(errRead)
+	redirect_stderr(originalSTDERR)
+	redirect_stdout(originalSTDOUT)
+
+	errstr = join(map(x->Char(x), errors))
+	outstr = join(map(x->Char(x), output))	
+	# Remove the manually added \n from stdout and stderr
+	return (chop(outstr), chop(errstr), result)
+end
+
+port = 32744 #TODO - find generic way to get an unused port.
+html = "<html>success</html>"
+html_len = length(html)
+server_running = true
+response = """
+HTTP/1.1 200 OK
+Content-Length: $html_len
+Content-Type: text/html
+Connection: Closed
+
+$html"""
+
+function start_server()
+	server_running = true
+	@async begin
+		server = listen(port)
+		while server_running
```

Comment:
You need to declare `server_running` as global on [line 43](https://github.com/JuliaLang/julia/pull/15930/files#diff-125a8ef99a2617e382db614c9a77673fR43) and [line 56](https://github.com/JuliaLang/julia/pull/15930/files#diff-125a8ef99a2617e382db614c9a77673fR56). You can also delete [line 33](https://github.com/JuliaLang/julia/pull/15930/files#diff-125a8ef99a2617e382db614c9a77673fR33)